pmm-app/tests/ia/pages/ruleTemplatesPage.js

```diff
@@ -0,0 +1,86 @@
+const { I } = inject();
+const fs = require('fs');
+const YAML = require('yaml');
+
+module.exports = {
+  url: 'graph/integrated-alerting',
+  columnHeaders: ['Name', 'Source', 'Created', 'Actions'],
+  elements: {
+    addedTemplate: '//td[text()="TemplateForAutomation"]/following-sibling::td[text()="User-defined (UI)"]',
+    ruleTemplateTab: '//li[@aria-label="Tab Alert Rule Templates"]',
+    templatesTableHeader: '$alert-rule-templates-table-thead',
+    templatesTable: '$alert-rule-templates-table-tbody',
+    templateName: '//tr/td[1]',
+    modalHeader: '$modal-header',
+    popUpTitle: '.alert-title',
+    columnHeaderLocator: columnHeaderText => `//th[text()="${columnHeaderText}"]`,
+  },
+  buttons: {
+    openAddTemplateModal: '$alert-rule-template-add-modal-button',
+    uploadFile: '$alert-rule-template-upload-button',
+    closeModal: '$modal-close-button',
+    addTemplate: '$alert-rule-template-add-button',
+    cancelAdding: '$alert-rule-template-cancel-button',
+    // editButtonBySource returns Edit template button locators for a given source
+    editButtonBySource: source => `//tr[descendant::td[contains(text(), "${source}")]]//button[@data-qa="edit-template-button"]`,
+  },
+  fields: {
+    templateInput: '$yaml-textarea-input',
+    fileInput: '//div[@data-qa="modal-content"]//input[@type="file"]',
+  },
+  messages: {
+    modalHeaderText: 'Add Alert Rule Template',
+    successfullyAdded: 'Alert rule template successfully added',
+    successfullyEdited: 'Alert rule template successfully added',
+    failedToParse: 'Failed to parse rule template.',
+    duplicateTemplate: id => `Template with name "${id}" already exists.`,
+  },
+  templateSources: {
+    ui: 'User-defined (UI)',
+    builtin: 'Built-in',
+  },
+  ruleTemplate: {
+    // templateNameAndContent parses yaml file and returns
+    // template name, full content as string and template id
+    templateNameAndContent: (ymlPath) => {
+      const content = fs.readFileSync(ymlPath, 'utf8');
+
+      try {
+        const name = YAML.parse(content).templates[0].summary;
+        const id = YAML.parse(content).templates[0].name;
```

Comment:
this is a bit confusing to me, name is being assigned to id, the summary is the name, is this expected or?

Reply:
Yeah, It is confusing. But that is how it named now. `summary` - Human-readable name that is shown in the List under `Name` column. `name` here is basically `id`

Reply:
okay makes sense, no need to move.